fastestimator/trace/metric/bleu_score.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Iterable, Union
+
+import numpy as np
+from nltk.translate.bleu_score import corpus_bleu
+
+from fastestimator.trace.meta.per_ds import per_ds
+from fastestimator.trace.trace import Trace
+from fastestimator.util.data import Data
+from fastestimator.util.traceability_util import traceable
+
+
+def get_hypotheses(input_data):
+    return [str(i) for i in input_data if i != 0]
+
+
+def get_reference(input_data):
+    return [get_hypotheses(input_data)]
+
+
+@per_ds
+@traceable()
+class BleuScore(Trace):
+    """Calculate the Bleu score for a nlp task and report it back to the logger.
```

Comment:
can we add a link here in the docs to somewhere that explains what Bleu score is? See MCC trace for example